### ISSUE TITLE
Feature Request: change unit in directive

### ISSUE THREAD

**Issue by meta-space** [hansmaad/ng-units#19]
Some units, e.g. Temperature, are global throughout the application.
Others like time, length, etc. may differ within the same page.

It would be great if it were possible to specify the unit on the ngQuantity directive, like ngQuantity="time:s"
This setting should be local and shouldn't change the unit in the global SystemOfUnits

**Comment by hansmaad** [comment]
Good idea, but I think it's better to have a separate binding for the unit like
```
<input ngQuantity="time" unit="s" >
<input ngQuantity="time" [unit]="myTimeUnit" >
<input [ngQuantity]="myQuantity" [unit]="myUnit" >
{{ value | ngQuantity:'time':'s' }}
```

**Comment by meta-space** [comment]
Coincidentally that is what I ended up doing in my custom quantity directive. 

However, I fear my implementation may no be optimal. I retrieve the Quantity instance from the SystemOfUnits.
If the unit needs to be changed, I create a local copy of Quantity & then call SelectUnit on the copy.
This way the SystemOfUnits is never affected by any directive, but I am copying Quantities around.

     private initQuantity(): void {
         this.unsubscribe();

         let quantity = typeof this.quantityAttr === 'string' ? this.system.get(this.quantityAttr) : this.quantityAttr;
         if (this.ngUnit) {
           const unit = quantity.findUnit(this.ngUnit as string);
           if (quantity.unit !== unit) {
             quantity = Object.assign(new Quantity(), quantity);
             quantity.selectUnit(unit);
           }
         }
         this.quantity = quantity;
         this.subscribe();
         this.updateUnit();
     }
